dynamic_programming/fibonacci_sequence_recursion_with_memoization.py

```diff
@@ -0,0 +1,48 @@
+# Fibonacci Sequence Using Recursion and Memoization
+
+# Memoization Concept: https://en.wikipedia.org/wiki/Memoization
+# Fibonacci Concept : https://pt.wikipedia.org/wiki/Sequ%C3%AAncia_de_Fibonacci
+
+cache: dict = dict()
+
+
+def fibonacci(n: int) -> int:
```

Comment:
```suggestion



def fibonacci(n: int, cache: dict = dict()) -> int:
```

Reply:
I really loved this suggestion, much better than declaring a dictionary in a global scope!